agent/consul/server_test.go

```diff
@@ -2107,6 +2107,70 @@ func TestServer_hcpManager(t *testing.T) {
 
 }
 
+func TestServer_addServerTLSInfo(t *testing.T) {
+	testCases := map[string]struct {
+		expectErr   bool
+		tlsConfig   tlsutil.Config
+		checkStatus func(*testing.T, hcpclient.ServerStatus)
+	}{
+		"Success": {
+			tlsConfig: tlsutil.Config{
+				InternalRPC: tlsutil.ProtocolConfig{
+					CAFile:               "../../test/ca/root.cer",
+					CertFile:             "../../test/key/ourdomain.cer",
```

Comment:
Sure! I was thinking that there was a time crunch and the tradeoff wouldn't be worth it. Since that's not the case, happy to figure out how to generate a file with intermediates

Reply:
We chatted a bit about this offline. I ended up duplicating the CertFile and then appending the cert for its issuer. This exercises the intermediate path but ends up showing duplicate CAs in the request. We talked about handling deduplication on the cloud side of things